### PR TITLE
[FIX] stock_by_warehouse: Scrollbar bug on stock_by_warehouse widget

### PR DESCRIPTION
**Pull request by agarzaarvizu**
### This MR refers to the ticket [#17884](https://www.vauxoo.com/web#id=17884&action=1809&model=helpdesk.ticket&view_type=form&cids=1&menu_id=1431)

Problem
- Scrollbar hidden and no height limit when the product is on several warehouses
image

Solution
- Import class to add overflow and height to the div on the scss file
Screenshot from 2022-02-22 18-42-47

### CODE REVIEW

**Comment by agarzaarvizu** [comment]
@andreaarce @imanie383 Please review.

cc @hugho-ad @desdelinux

**Comment by desdelinux** [comment]
Where can I test functionally? @agarzaarvizu @hugho-ad

File: stock_by_warehouse/static/src/scss/main.scss
Comment (imanie383) [inline stock_by_warehouse/static/src/scss/main.scss]: Why do you need to import this?

The next image is an odoo14 installation and this scss is already import
image

File: stock_by_warehouse/static/src/scss/main.scss
Comment (imanie383) [inline stock_by_warehouse/static/src/scss/main.scss]: popover-body is a `common` class. Maybe you should make the selector more specific

Aren't you modifying other elements?

image

File: stock_by_warehouse/static/src/scss/main.scss
Comment (luisg123v) [inline stock_by_warehouse/static/src/scss/main.scss]: Are you sure you need to rely on enterprise? Why? This module doesn't depend on any enterprise module

**Review by luisg123v — CHANGES_REQUESTED** [review]
- PR's Description is OK, but title doesn't mention what's being fixed
- commit description is empty. Remember the one that's preserved in the history is the commit

File: stock_by_warehouse/static/src/scss/main.scss
Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/scss/main.scss]: @luisg123v You were right, I've changed the import file to an addons from odoo, please check again, also the PR title.

Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/scss/main.scss]: The problem was with our widget, there wasn't working the overflow and hasn't a height limit, so we decided to add the import and fix it in case that module is needed.

File: stock_by_warehouse/static/src/scss/main.scss
Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/scss/main.scss]: I've changed the import file, I think that's the right one, or which file and class you think is better to import??

Comment (luisg123v) [reply stock_by_warehouse/static/src/scss/main.scss]: This module doesn't depend on website, neither.

Comment (imanie383) [reply stock_by_warehouse/static/src/scss/main.scss]: You shouldn't import anything, odoo base already has the base styles
Could you share an instance to analyze the problem?

Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/scss/main.scss]: I removed the import, is working correctly, please check.

File: stock_by_warehouse/static/src/scss/main.scss
Comment (imanie383) [inline stock_by_warehouse/static/src/scss/main.scss]: **Add a new class or div and changed it instead .popover-body**
image

otherwise, you are going to modify all popovers to 400px
https://getbootstrap.com/docs/4.0/components/popovers/
image

**Review by imanie383 — CHANGES_REQUESTED** [review]

File: stock_by_warehouse/static/src/scss/main.scss
Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/scss/main.scss]: Done, please review

**Comment by desdelinux** [comment]
Funtional Review is ok

LGTM @agarzaarvizu 

cc @hugho-ad

File: stock_by_warehouse/static/src/xml/template.xml
Comment (luisg123v) [inline stock_by_warehouse/static/src/xml/template.xml]: The more specific is the name, the less likely is having collisions, e.g.:

```suggestion
        <div class="panel clearfix container stock-by-warehouse-widget">
```

**Review by luisg123v — CHANGES_REQUESTED** [review]
Just a small comment

**Comment by imanie383** [comment]
image

IMP: https://youtu.be/7RdvijatPGw

**Review by imanie383 — CHANGES_REQUESTED** [review]
image

IMP: https://youtu.be/SbkFTx-qd9s

https://getbootstrap.com/docs/4.0/components/popovers/
Change popover `template` option to add your new class beside .popover-body

**Review by imanie383 — CHANGES_REQUESTED** [review]
.

**Comment by agarzaarvizu** [comment]
@luisg123v @imanie383 Done

cc @desdelinux @hugho-ad

**Review by hugho-ad — APPROVED** [review]

**Comment by hugho-ad** [comment]
LGTM :+1:

File: stock_by_warehouse/static/src/js/widget.js
Comment (luisg123v) [inline stock_by_warehouse/static/src/js/widget.js]: I think this is too complex to be harcoded as a string, you could use `qweb.render("template_name")` and move this to a template

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: stock_by_warehouse/static/src/js/widget.js
Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/js/widget.js]: as I understood and following the @imanie383 recommendations, this is the best way, at least enough to solve the issue, or what do you think @imanie383?

Comment (imanie383) [reply stock_by_warehouse/static/src/js/widget.js]: @luisg123v  is right, It's just create a static template

**Comment by agarzaarvizu** [comment]
@luisg123v @imanie383 Done, please review

cc @desdelinux @hugho-ad

**Comment by imanie383** [comment]
LGTM +1

Maybe you can omit </> in the template, but is Optional

**Review by luisg123v — APPROVED** [review]

**Comment by luisg123v** [comment]
Waiting for @imanie383's approval to merge.

**Comment by imanie383** [comment]
@luisg123v Done

File: stock_by_warehouse/static/src/xml/template.xml
Comment (luisg123v) [inline stock_by_warehouse/static/src/xml/template.xml]:
```suggestion
            <div class="popover-body stock-by-warehouse-widget" />
```

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: stock_by_warehouse/static/src/xml/template.xml
Comment (agarzaarvizu) [reply stock_by_warehouse/static/src/xml/template.xml]: Done

**Review by luisg123v — APPROVED** [review]

**Pull request merged**
